src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/CaseOfficerPersonalisationFactory.java

```diff
@@ -36,4 +48,67 @@ public CaseOfficerPersonalisationFactory(
                 .put("Hyperlink to user’s case list", iaCcdFrontendUrl)
                 .build();
     }
+
+    public Map<String, String> createListedCase(
+        AsylumCase asylumCase
+    ) {
+        requireNonNull(asylumCase, "asylumCase must not be null");
+
+        final HearingCentre listedHearingCentre =
+            asylumCase
+                .read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)
+                .orElseThrow(() -> new IllegalStateException("listCaseHearingCentre is not present"));
+
+        final String hearingCentreAddress =
+            stringProvider
+                .get("hearingCentreAddress", listedHearingCentre.toString())
+                .orElseThrow(() -> new IllegalStateException("hearingCentreAddress is not present"));
+
+        final String hearingDateTime =
+            asylumCase
+                .read(AsylumCaseDefinition.HEARING_DATE_TIME, String.class)
+                .orElseThrow(() -> new IllegalStateException("hearingDateTime is not present"));
+
+        return
+            ImmutableMap
+                .<String, String>builder()
```

Comment:
Can probably infer the generic types here.

Reply:
Done

Reply:
Just saying that you could do possibly do something like `ImmutableMap.<>` instead of `ImmutableMap<String, String>`.  Not essential as it works the same, just a suggestion.

Reply:
Okay, thanks Ravi.